### DISCUSSION TITLE
DeepWannier/DeepDipole NN not learning data at all? Inference is very wrong.

### DISCUSSION THREAD

**Discussion by alikhamze** [deepmodeling/deepmd-kit#2773] [Q&A] [answered]
Hello,

I've been working on training a DeepWannier model for some time now and have been struggling to get accurate predictions. 

As a test, to verify whether it was the training data or DeepWannier itself that was the core issue, I created an artificial dataset where each structure is made by randomly perturbing every atom of the crystal with small amplitude perturbations (0.01, 0.05, and 0.075 angstroms), and where I set every atomic dipole moment to mu = (0.005, 0, 0) and where the global dipole moment is the exact sum of the atomic dipole moments. 
In other words, I was trying to set up the NN to predict the same atomic and global dipole moment for every structure, to make sure it is learning and can infer correctly.

After using the trained model to infer on the validation set, there is no correlation between the predicted values and the inferred values. 
<img src="https://github.com/deepmodeling/deepmd-kit/assets/6139089/61dac56d-30d4-4e59-b9e1-562146085469" alt="drawing" width="400"/>
It appears that DW NN is not learning the training data at all...

I've attached a tar.gz file containing the model training inputs and outputs (I deleted the frozen and compressed models for github file size limits), the training data (I removed 60% of it to reduce the file size, but it should still be helpful if the issue lies there), and code for inference from the frozen, compressed model. Training and inference were done using DeepMD 2.2.1.

Is there some obvious problem in my training data, model, or inference? Or something else that I am missing? 

Thank you,
-Ali

[bad_data.tar.gz](https://github.com/deepmodeling/deepmd-kit/files/12488926/bad_data.tar.gz)
[deepwannier_model_inference.tar.gz](https://github.com/deepmodeling/deepmd-kit/files/12488928/deepwannier_model_inference.tar.gz)

**Comment by wanghan-iapcm** [comment, marked as answer]
The deep dipole model, which is rotationally equivalent cannot learn your data, which is rotatinally invariant.

**Reply by alikhamze** [reply]
Thanks Han! I was really stumped by this result, that explains it.
I tested with less pathological data set and it worked much better. I might have some other questions about improving model accuracy but will make another discussion if needed.

Best,
-Ali
